web/client/selectors/widgets.js

```diff
@@ -46,20 +47,34 @@ const getWidgetAttributeFilter = (id, attributeName) => createSelector(
         return widget && widget.quickFilters && widget.options && find(widget.options.propertyName, f => f === attributeName) && widget.quickFilters[attributeName] || {};
     });
 
+const getWidgetById = (id) => createSelector(
+    getVisibleFloatingWidgets,
+    (widgets) => {
+        const widget = find(widgets, {id});
+        return widget;
+    });
+
 const getCollapsedIds = createSelector(
     getCollapsedState,
     (collapsed = {}) => Object.keys(collapsed)
 );
 const getMapWidgets = state => (getFloatingWidgets(state) || []).filter(({ widgetType } = {}) => widgetType === "map");
+const getTableWidgets = state => (getFloatingWidgets(state) || []).filter(({ widgetType } = {}) => widgetType === "table");
 
 /**
  * Find in the state the available dependencies to connect
  */
 const availableDependenciesSelector = createSelector(
     getMapWidgets,
+    getTableWidgets,
     mapSelector,
-    (ws = [], map = []) => ({
-        availableDependencies: ws.map(({id}) => `widgets[${id}].map`).concat(castArray(map).map(() => "map"))
+    pathnameSelector,
+    (ws = [], tableWidgets = [], map = [], pathname) => ({
+        availableDependencies:
+            ws
+                .map(({id}) => `widgets[${id}].map`)
+                .concat(castArray(map).map(() => "map"))
+                .concat(castArray(tableWidgets).filter(() => pathname.indexOf("viewer") === -1).map(({id}) => `widgets[${id}]`))
```

Comment:
Because there was a problem during selection since the table widgets were inside the map and not outside as in the dashboard, so i disabled this possibility in the viewer as explained in the PR description